File: 0101_0200/0164/0164.py
def sq(n):
    while n > 9:
        s = 0
        while n > 0:
            s += n % 10
            n //= 10
        n = s
    return n
 
with open("input.txt", "r") as f, open("output.txt", "w") as q:
    s = [int(x) for x in list(f.readline().strip())]
    r = False
    if len(s) > 1:
        a, b = s[0], sum(s[1:])
        r = (sq(a) == sq(b))
        if not r:
            for i in range(1, len(s)):
                a += s[i]
                b -= s[i]
                r = (sq(a) == sq(b))
                if r:
                    break
    q.write("YES" if r else "NO")

File: 0101_0200/0164/0164.c
#include <stdio.h>
#include <string.h>

int sq(int n)
{
	while(n > 9)
	{
		int a = 0;
		while (n > 0)
			a += n % 10, n /= 10;
		n = a;
	}
	return n;
}

int main()
{
	FILE *f = fopen("input.txt", "r"), *q = fopen("output.txt", "w");
	int l;
	char s[101], r = 0;

	fscanf(f, "%s", s);
	if ((l = strlen(s)) > 1)
	{
		int a = s[0] - '0', b = 0;
		for (int i = 1; i < l; ++i)
			b += s[i] - '0';
		r = (sq(a) == sq(b));
		for (int i = 1; !r && i < l; ++i)
			r = (sq(a += s[i] - '0') == sq(b -= s[i] - '0'));
	}
	fprintf(q, r ? "YES" : "NO");
	return 0;
}
